app/knights_battle/battle.py

```diff
@@ -0,0 +1,26 @@
+from app.knights_battle.knight import Knight
+
+
+def battle(knights_config: dict) -> dict:
+    result = {}
+
+    for element in knights_config:
+        knight = Knight(name="", power=0, hp=0, protection=0)
+        knight.adding_value_to_instance(knights_config[element])
+        knight.adding_potion_to_instance(knights_config[element])
+        result[knight.name] = knight

```

Comment:
To make this code more readable you can split it by methods.
for example `apply_armour(). apply_weapon(), apply_potion()`